gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -318,11 +318,27 @@ public GoogleCloudStorageImpl(
       throws IOException {
     this(
         options,
-        createStorage(options, httpRequestInitializer),
+        createStorage(
+            options,
+            new GcsioTrackingHttpRequestInitializer(httpRequestInitializer, httpStatistics)),
         /* credentials= */ null,
         downscopedAccessTokenFn);
   }
 
+  private static HttpRequestInitializer setGcsRequestTracker(
+      GoogleCloudStorageOptions options, Credential credential) {
+    HttpRequestInitializer httpRequestInitializer =
+        new RetryHttpInitializer(credential, options.toRetryHttpInitializerOptions());
+    gcsRequestsTracker =
+        new GcsioTrackingHttpRequestInitializer(httpRequestInitializer, httpStatistics);
+    if (options.isGrpcEnabled()) {
+      return (HttpRequestInitializer)
+          new RetryHttpInitializer(
+              gcsRequestsTracker, credential, options.toRetryHttpInitializerOptions());
+    }
+    return (HttpRequestInitializer) gcsRequestsTracker;
```

Comment:
This can be simplified, For grpc, i don't see a need to wrap twice 

retry -> gcsiotracker -> retry initializer

Reply:
Updated

Reply:
seems like this is not updated

Reply:
Updated